impeller/renderer/command.h

```diff
@@ -28,13 +28,26 @@ namespace impeller {
 template <class T>
 struct Resource {
   using ResourceType = T;
-  const ShaderMetadata* isa;
   ResourceType resource;
 
-  Resource() : isa(nullptr) {}
+  Resource() : isa_(nullptr) {}
 
-  Resource(const ShaderMetadata* p_isa, ResourceType p_resource)
-      : isa(p_isa), resource(p_resource) {}
+  Resource(const ShaderMetadata* isa, ResourceType p_resource)
+      : resource(p_resource), isa_(isa) {}
+
+  Resource(std::shared_ptr<const ShaderMetadata>& isa, ResourceType p_resource)
+      : resource(p_resource), isa_(nullptr), dynamic_isa_(isa) {}
+
+  const ShaderMetadata* isa() const {
```

Comment:
Perhaps just call this `GetMetadata` and change the ivar to `metadata_`. Admittedly, the `isa` ptr is a holdover from the Objective-C runtime and its not particularly descriptive if you haven't done some runtime mangling.

Reply:
I like the technique though.

Reply:
done